### DISCUSSION TITLE
How to change specific icons? (perhaps by icon name?)

### DISCUSSION THREAD

**Discussion by MoltenMonster** [kyazdani42/nvim-web-devicons#508] [Q&A]
I want to replace 󰲡 (a 1 inside a circle), you can search `nf-md-numeric_1_circle_outline` on [nerdfonts.com](https://nerdfonts.com) to see it. I would replace it with an Apple [SF Symbol](https://developer.apple.com/sf-symbols/) instance 1.circle:

![CleanShot 2024-10-24 at 19 39 43@2x](https://github.com/user-attachments/assets/2ccbae34-5fd0-48b7-8f26-e6ed575fccbe)

I can paste its text character 􀀺 into a buffer inside Neovide and it shows without any extra steps or setup.

**Comment by alex-courtis** [comment]
You can override existing icons and add your own, see [setup](https://github.com/nvim-tree/nvim-web-devicons?tab=readme-ov-file#setup)

Use `:NvimWebDeviconsHiTest` to see all icons and overrides.

e.g.
```lua
require("nvim-web-devicons").setup({
  -- override *.xml extension
  override_by_extension = {
    ["xml"] = {
      icon = "󰲡",
      color = "#00afff",
      cterm_color = "39",
      name = "Xml",
    },
  },
  -- override .gitignore name
  override_by_filename = {
    [".gitignore"] = {
      icon = "󰲡",
      color = "#20c2e3",
      cterm_color = "45",
      name = "Gitignore"
    }
  },
  -- create new extensions
  override = {
    ["foo"] = {
      icon = "󰲡",
      color = "#428850",
      cterm_color = "65",
      name = "Foo"
    }
  },
})
```

**Reply by MoltenMonster** [reply]
@alex-courtis Yeah I saw that but can you give an example for the icon I'm trying to change: 󰲡 (nf-md-numeric_1_circle_outline). I couldn't figure it out, for instance I tried changing both `["foo"]` and `"Foo"` to `nf-md-numeric_1_circle_outline` but that doesn't work.

**Reply by alex-courtis** [reply]
Those examples do use the circle-1 icon - I just copied that icon character and pasted it into the icon field.

Were you perhaps using `icon = "nf-md-numeric_1_circle_outline",` ?

I'm using a nerd font in my browser and it shows the icon:
![20241106_094648](https://github.com/user-attachments/assets/27036cb8-4e64-42c1-9928-27eb7772e043)

**Reply by MoltenMonster** [reply]
@alex-courtis I should've mentioned, I'm not trying to override file extensions, I want it so that whenever the nerd icon 󰲡 shows (like if a plugin with nvim-web-devicons as a dependency shows the icon), it shows SF symbol instead. I guess I should've specified that.

**Reply by MoltenMonster** [reply]
@alex-courtis Nevermind, I thought nvim-web-devicons was for serving any icons (at least nerd font ones) but its not, its only for subsets like file types and programming languages.
